editor/modes/visual-editor/block.js

```diff
@@ -148,6 +167,8 @@ class VisualEditorBlock extends wp.element.Component {
 				onMouseMove={ this.maybeHover }
 				onMouseLeave={ onMouseLeave }
 				className={ className }
+				data-type={ block.blockType }
+				{ ...getBlockAttributesAsProps( block.attributes ) }
```

Comment:
I bounced back and forth on a few names, or whether this could somehow be integrated into the `edit` function itself (via callback, or a special shaped return value including metadata).

How does `getEditWrapperProps` sound to you?